Don't print error log for indexLookUpExecutor when context canceled

## Enhancement
When indexLookUpExecutor is closed before its index/table worker finished(it may be caused by limit operator), it will use cancelFunc to inform index/tableWork. Currently, this will lead to useless error logs: "[2025/05/12 10:25:18.142 +08:00] [ERROR] [distsql.go:1572] ["table reader fetch next chunk failed"] [conn=2120221196] [session_alias=] [error="context canceled"]"
1. LimitExec returns nil when fetches enough rows, and its Close function will be called next:
https://github.com/pingcap/tidb/blob/2503d50c4d14f90862c7603652d5e4c90b70cb0c/pkg/executor/select.go#L429-L431
2. When indexLookUpExecutor's Close function is called, it will call its cancelFunc to inform index/tableWork:
https://github.com/pingcap/tidb/blob/2503d50c4d14f90862c7603652d5e4c90b70cb0c/pkg/executor/distsql.go#L898-L901
3. In coprocessor, each time it tries to receive response, it checks the context status, and return error when context canceled:
https://github.com/pingcap/tidb/blob/2503d50c4d14f90862c7603652d5e4c90b70cb0c/pkg/store/copr/coprocessor.go#L1130-L1134
4. In indexLookUpExecutor's tableWork, it outputs error log when errors returned from coprocessor:
https://github.com/pingcap/tidb/blob/2503d50c4d14f90862c7603652d5e4c90b70cb0c/pkg/executor/distsql.go#L1564-L1567

--- a/pkg/executor/distsql.go
+++ b/pkg/executor/distsql.go
@@ -869,7 +869,9 @@ func (e *IndexLookUpExecutor) buildTableReader(ctx context.Context, task *lookup
 	tableReaderExec.buildVirtualColumnInfo()
 	tableReader, err := e.dataReaderBuilder.buildTableReaderFromHandles(ctx, tableReaderExec, task.handles, true)
 	if err != nil {
-		logutil.Logger(ctx).Error("build table reader from handles failed", zap.Error(err))
+		if ctx.Err() != context.Canceled {
+			logutil.Logger(ctx).Error("build table reader from handles failed", zap.Error(err))
+		}
 		return nil, err
 	}
 	return tableReader, nil
@@ -1531,7 +1533,9 @@ func (w *tableWorker) executeTask(ctx context.Context, task *lookupTableTask) er
 	tableReader, err := w.idxLookup.buildTableReader(ctx, task)
 	task.buildDoneTime = time.Now()
 	if err != nil {
-		logutil.Logger(ctx).Error("build table reader failed", zap.Error(err))
+		if ctx.Err() != context.Canceled {
+			logutil.Logger(ctx).Error("build table reader failed", zap.Error(err))
+		}
 		return err
 	}
 	defer func() { terror.Log(exec.Close(tableReader)) }()
@@ -1562,7 +1566,9 @@ func (w *tableWorker) executeTask(ctx context.Context, task *lookupTableTask) er
 		chk := exec.TryNewCacheChunk(tableReader)
 		err = exec.Next(ctx, tableReader, chk)
 		if err != nil {
-			logutil.Logger(ctx).Error("table reader fetch next chunk failed", zap.Error(err))
+			if ctx.Err() != context.Canceled {
+				logutil.Logger(ctx).Error("table reader fetch next chunk failed", zap.Error(err))
+			}
 			return err
 		}
 		if chk.NumRows() == 0 {

